docs: add Responsible AI Transparency Note (#1594)
microsoft/hve-core: TRANSPARENCY-NOTE.md

## The basics of HVE Core

### Introduction

HVE Core is a static corpus of prompt-engineering and agentic-customization artifacts: custom agents, prompts, instructions, skills, collections, PowerShell scripts, GitHub Actions workflows, and a Visual Studio Code extension. Its purpose is to give engineering teams a curated, governance-aware starting point for AI-assisted software work that runs on top of GitHub Copilot.

HVE Core does not train models, does not host inference, does not call external services at runtime, and does not process personally identifiable information within its own boundaries. Inference happens entirely on the host platform (GitHub Copilot Chat or GitHub Copilot CLI). HVE Core's Responsible AI exposure is concentrated in three areas:

1. What the artifacts say to and about the downstream model that the host platform invokes.

> 2. The trust adopters place in artifacts that carry Microsoft branding through the `microsoft/hve-core` repository and the official Visual Studio Code extension.

Reviewer: "visual studio code GHCP extension"

3. The persistent memory layer and the customer-handoff workflows that some agents orchestrate.

> This Transparency Note describes the system at the level of the umbrella repository. Per-agent appendices at the end of this document add specifics for the highest-impact governance-bearing agents and the one direct generative-AI touchpoint (the Customer Card Render skill).

Reviewer: probably need to talk about the phase gated release system here too

| Prompt               | A reusable user-message template (`*.prompt.md`) intended to be loaded into a Copilot Chat or CLI session.                                                                                         |
| Instructions         | Behavioral guidance (`*.instructions.md`) that an agent or session loads to shape the model's responses for a given file scope, language, or workflow.                                             |
| Skill                | A self-contained capability package (`SKILL.md` plus optional scripts and references) that documents a reusable domain capability.                                                                 |
| Collection           | A bundled set of artifacts (`*.collection.yml` plus a companion markdown description) that a publisher can install as a unit through one of the distribution channels.                             |
| Subagent             | A custom agent invoked by another agent for a focused task (typical examples: a researcher subagent for read-only investigation, an implementer subagent for a single phase of work).                |

> | Distribution channel | One of six routes by which HVE Core artifacts reach an adopter: VS Code extension, npm package, GitHub plugin marketplace, direct git clone, internal Microsoft mirror, customer-handoff repository. |

Reviewer: we don't yet support the NPM Package distro for GHCP CLI ingest ... it's been requested but it's probably a maintenance bridge too far at the moment. There's also not internal mirror and no handoff repository that's public.

| Host platform        | The Copilot surface that performs the actual inference: GitHub Copilot Chat in Visual Studio Code, or GitHub Copilot CLI. HVE Core does not include or replace this platform.                        |
| Memory layer         | The persistent notes feature exposed by the host platform. HVE Core agents may write user-scope, session-scope, or repository-scope memory. Storage and retention are controlled by the host.        |
| Governance-bearing   | An adjective applied to agents whose outputs influence downstream decisions in a binding-by-default way (planning agents, code-review agents that gate pull requests, customer-handoff agents).      |

## Capabilities

### System behavior

HVE Core ships static text artifacts and supporting tooling. When an adopter loads an HVE Core artifact into the host platform, three things happen:

> 1. The host platform reads the artifact contents into its session context.

Reviewer: ```suggestion
1. The host platform reads applicable artifacts contents into its session context.
```

2. The user interacts with the host platform's model, which is shaped by the artifact's instructions.
3. The model's responses are returned to the user through the host platform's standard surface.

> HVE Core itself has no inference runtime, no API endpoints, no external network calls during artifact authoring or distribution, and no telemetry pipeline. Validation tooling (linters, frontmatter checks, Pester tests, plugin generation) runs in CI on pull requests; nothing runs on adopter machines unless the adopter explicitly invokes a packaged script or installs the Visual Studio Code extension.

Reviewer: ```suggestion
HVE Core itself has no inference runtime, no API endpoints, and has no influence over user authorization of external network calls during artifact authoring or distribution, and no telemetry pipeline. Validation tooling (linters, frontmatter checks, Pester tests, plugin generation) runs in CI on pull requests; nothing runs on adopter machines unless the adopter explicitly invokes a packaged script or installs the Visual Studio Code extension.
```

A small set of artifacts crosses the boundary into direct generative output:

> * The **Customer Card Render** skill drives a synthetic-persona slide-rendering pipeline that uses image generation. This is the only direct generative-AI touchpoint that ships in HVE Core. Because the output is synthetic media depicting people-like figures, the skill enforces a low-fidelity visual style and is documented separately in Appendix 5 alongside its AI-disclosure, redaction, and stereotyping-review controls.

Reviewer: this isn't really true and we need to clarify its classification ... we should also default to generating a test prompt that user can use to get this capability via other platforms

> * The **PowerPoint Builder** and **TTS Voice-over** experimental skills produce slide decks and voice audio from authored YAML, but they do not generate likenesses of people or claim to be the speaker; they are conventional content-assembly tools that consume model output authored elsewhere.

Reviewer: Probably need to highligh work iq integrations, and how TTS requires a MSFT subscription for clip champ, etc.

#### Responsibility boundary

| Aspect                         | HVE Core (microsoft/hve-core)                                              | Host platform (Copilot Chat or CLI)                                            |
| ------------------------------ | -------------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
| Inference                      | None                                                                       | All inference happens here                                                     |
| Model selection and management | Recommends models in artifact frontmatter; cannot enforce a model choice   | Selects, hosts, and operates the model                                         |
| Safety classifiers             | None                                                                       | Provides input and output classification, jailbreak detection, content filters |

> | User authentication            | None                                                                       | Manages identity, tenant scope, and access controls                            |

Reviewer: This is a grey area and we should be exlict that some skills will support authentication to 3P services. All skills include a skill spcific threat model, etc.

> | Telemetry and feedback         | None at the artifact level; CI logs only                                   | Collects user interaction telemetry per the host's privacy statement           |

Reviewer: cross link the VSCode telemetry and usage collection docs ... we have no influence over this, but we should patch to them

| Persistent memory              | Authors agents that write to memory; does not store memory itself          | Stores, retains, and exposes the memory layer                                  |
| Attribution                    | Embeds "Brought to you by microsoft/hve-core" attribution where convention | Surfaces attribution to the user through the chat session                      |

### Use cases

#### Intended uses

HVE Core is designed to be adopted in these scenarios:

> * **Engineering productivity scaffolding for Microsoft engineers and field teams.** Agents and prompts that codify common engineering workflows (research, planning, implementation, review, discovery) reduce the activation cost of starting a new task with Copilot.

Reviewer: ```suggestion
* **Engineering and SLDC productivity scaffolding for sofware developers.** Agents and prompts that codify common engineering workflows (research, planning, implementation, review, discovery) reduce the activation cost of starting a new task with Copilot.
```

* **Coding-standards enforcement through Copilot.** Per-language instruction files and code-review agents bring consistent guidance into Copilot sessions when adopters work in C#, Python, PowerShell, Rust, Bash, Bicep, Terraform, or related ecosystems.

> * **Governance-aware planning support.** The RAI Planner, Security Planner, and SSSC Planner agents help teams structure assessments aligned with the NIST AI RMF, STRIDE, and the OpenSSF Scorecard family. The agents produce drafts; qualified humans must review and validate every output.

Reviewer: ```suggestion
* **Governance-aware planning support.** The RAI Planner, Security Planner, and SSSC Planner agents help teams structure assessments aligned with open standards such as the NIST AI RMF, STRIDE, and the OpenSSF Scorecard family. The agents produce drafts; qualified humans must review and validate every output.
```

* **Backlog management assistance for Azure DevOps, GitHub Issues, and Jira.** Agents can search, draft, triage, and prepare backlog updates for human review and approval.

> * **Design Thinking facilitation in customer engagements.** Coaching agents support a structured Design Thinking workflow; the Customer Card Render skill produces synthetic personas for stakeholder communication subject to the constraints in Appendix 5.

Reviewer: ```suggestion
* **Design Thinking facilitation.** Coaching agents support a structured Design Thinking workflow; the Customer Card Render skill produces synthetic personas for stakeholder communication subject to the constraints in Appendix 5.
```

* **Inference of personal attributes.** Agents and prompts must not be repurposed to infer protected characteristics of individuals from incidental data.
* **Surveillance.** HVE Core artifacts must not be used to monitor individuals without their knowledge and consent in jurisdictions that require it.
* **Synthetic media of real people without disclosure.** The Customer Card Render skill produces synthetic personas only and enforces a low-fidelity style. It must not be reconfigured to render likenesses of identifiable individuals or to omit the synthetic-media disclosure.
* **Jailbreak or guardrail evasion.** Artifacts must not be modified to suppress the host platform's safety controls.
* **High-stakes automated determinations.** Agent outputs are not calibrated for use as the sole basis of any decision with significant consequences for an individual.
* **Use on unsupported clients.** HVE Core is engineered for current GitHub Copilot Chat in Visual Studio Code and the GitHub Copilot CLI. Behavior on other clients, on outdated versions, or on third-party LLM hosts is not characterized.

> Legal and regulatory considerations: organizations must evaluate their own legal and regulatory obligations before using AI-related tooling. AI services may not be appropriate in all scenarios and must comply with applicable laws and the terms of service of the host platform.

Reviewer: probably want to link to AI COde of Conduct for GHCP, and it's assets in this space as well

3. **Pull-request review.** Human review is required for every artifact change. Dependency pinning, action SHA staleness, and supply-chain checks run as automated gates that surface to reviewers.
4. **Versioning and publication.** Releases follow `release-please` conventional-commit conventions with a CHANGELOG. The Visual Studio Code extension and npm packages carry version metadata that adopters can pin against.
5. **Issue and feedback channel.** GitHub issues on `microsoft/hve-core` are the primary channel for bug reports, feature requests, and concerns. The maintainers triage and route from there.

Performance against a specific downstream model is not measured by HVE Core. Adopters who need reproducible behavior should pin both the artifact version and the host platform configuration.

### Best practices for improving system performance

* **Pin to a known-good release tag.** Treat the canonical repository's main branch as a moving target. For production-relevant adoption, pin to a release tag and review changes before upgrading.

> * **Adopt one collection at a time.** HVE Core ships thirteen collections. Most adopters do not need all of them. Start with the collection closest to your workflow (for example, `coding-standards` for engineering productivity, `security` for threat modeling assistance, `rai-planning` for Responsible AI assessment scaffolding) and expand from there.

Reviewer: This number will change over time as we decrease the agent footprint and migrate to more skill usage. Consider removing the specific value and cross link to the collections manifest for current and accurate data.
